receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -266,22 +266,25 @@ func (h *headerReceiver) newContext(ctx context.Context, hdrs map[string][]strin
 }
 
 // logStreamError decides how to log an error.
-func (r *Receiver) logStreamError(err error, where string) {
+func (r *Receiver) logStreamError(err error, where string) (otelcodes.Code, string) {
 	var code codes.Code
 	var msg string
 	// gRPC tends to supply status-wrapped errors, so we always
 	// unpack them.  A wrapped Canceled code indicates intentional
 	// shutdown, which can be due to normal causes (EOF, e.g.,
 	// max-stream-lifetime reached) or unusual causes (Canceled,
 	// e.g., because the other stream direction reached an error).
+	occode := otelcodes.Unset
 	if status, ok := status.FromError(err); ok {
 		code = status.Code()
+		occode = otelcodes.Error
```

Comment:
It looks like this logic doesn't exactly line up with what's being added in `recvOne`. It looks like if the `status.Code()` is `codes.Canceled`, we don't want to consider it an error (at least in `recvOne). But if the status code is canceled, this method will consider it an error.

Is the difference intentional? Am I missing something, or maybe misunderstanding the mapping between error codes and errors?

Reply:
Good point. I will revise for consistency.

Reply:
I've added the consistency you were looking for, I think. Any canceled code will become an Unset span status. All other codes will become span errors.